src/uint256.h

```diff
@@ -125,20 +125,11 @@ class base_blob
 template <unsigned int BITS>
 consteval base_blob<BITS>::base_blob(std::string_view hex_str)
 {
-    // Non-lookup table version of HexDigit().
-    auto from_hex = [](const char c) -> int8_t {
-        if (c >= '0' && c <= '9') return c - '0';
-        // Only lowercase letters are supported, for consistency.
-        if (c >= 'a' && c <= 'f') return c - 'a' + 0xA;
-
-        throw "Called ctor with an invalid hex digit";
-    };
-
     if (hex_str.length() != m_data.size() * 2) throw "Hex string must fit exactly";
     auto str_it = hex_str.rbegin();
     for (auto& elem : m_data) {
-        auto lo = from_hex(*(str_it++));
-        elem = (from_hex(*(str_it++)) << 4) | lo;
+        auto lo = ConstevalHexDigit(*(str_it++));
```

Comment:
I'm unable to compile 09458eadc9a4484ba37a70d1b378ed3f3c9e31d0  using:

```
% clang --version
Homebrew clang version 17.0.6
Target: arm64-apple-darwin23.3.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
```


```
git clean -xdf && git checkout 09458eadc9a4484ba37a70d1b378ed3f3c9e31d0 && ./autogen.sh && ./configure --without-gui && make -j 7

./uint256.h:131:19: error: call to consteval function 'ConstevalHexDigit' is not a constant expression
        auto lo = ConstevalHexDigit(*(str_it++));
                  ^
./uint256.h:173:60: note: in instantiation of member function 'base_blob<256>::base_blob' requested here
    consteval explicit uint256(std::string_view hex_str) : base_blob<256>(hex_str) {}
                                                           ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/c++/v1/__iterator/reverse_iterator.h:168:56: note: read of non-constexpr variable 'str_it' is not allowed in a constant expression
    reverse_iterator operator++(int) {reverse_iterator __tmp(*this); --current; return __tmp;}
                                                       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX14.0.sdk/usr/include/c++/v1/__iterator/reverse_iterator.h:168:56: note: in call to 'reverse_iterator(str_it)'
./uint256.h:131:39: note: in call to '&str_it->operator++(0)'
        auto lo = ConstevalHexDigit(*(str_it++));
                                      ^
./uint256.h:129:10: note: declared here
    auto str_it = hex_str.rbegin();
```

Reply:
> What is your macOS and XCode version? IIRC Xcode 14 is no longer supported, see also https://github.com/bitcoin/bitcoin/pull/29934, but I think this isn't documented, nor tested, so probably up for debate.

Thanks, I'm on macOS 14.3.1 and just bumped XCode from 14.3.1.0.1.1683849156 to 15.3.0.0.1.1708646388 which resolves the issue 👍

Reply:
> So I guess this commit drops support for XCode on macOS Ventura 13. Not sure if doc/build-osx.md needs to be adjusted, similar to https://github.com/bitcoin/bitcoin/pull/29934/files

According to my earlier (pending and therefore out of order) message, https://github.com/bitcoin/bitcoin/pull/30377#discussion_r1716900164, the clang version is not the issue. So simply changing to `llvm@18` in the docs might not be correct.

What is your clang version after the XCode upgrade @stickies-v?